source/common/http/utility.cc

```diff
@@ -1287,5 +1333,73 @@ Http::Code Utility::maybeRequestTimeoutCode(bool remote_decode_complete) {
                                 : Http::Code::RequestTimeout;
 }
 
+std::string Utility::newUri(::Envoy::OptRef<const Utility::RedirectConfig> redirect_config,
+                            const Http::RequestHeaderMap& headers) {
+  absl::string_view final_scheme;
+  absl::string_view final_host;
+  absl::string_view final_port;
+  absl::string_view final_path;
+
+  if (redirect_config.has_value() && !redirect_config->scheme_redirect_.empty()) {
+    final_scheme = redirect_config->scheme_redirect_;
+  } else if (redirect_config.has_value() && redirect_config->https_redirect_) {
+    final_scheme = Http::Headers::get().SchemeValues.Https;
+  } else {
+    // Serve the redirect URL based on the scheme of the original URL, not the
+    // security of the underlying connection.
+    final_scheme = headers.getSchemeValue();
+  }
+
+  if (redirect_config.has_value() && !redirect_config->port_redirect_.empty()) {
+    final_port = redirect_config->port_redirect_;
+  } else {
+    final_port = "";
+  }
+
+  if (redirect_config.has_value() && !redirect_config->host_redirect_.empty()) {
+    final_host = redirect_config->host_redirect_;
+  } else {
+    ASSERT(headers.Host());
+    final_host = processRequestHost(headers, final_scheme, final_port);
+  }
+
+  std::string final_path_value;
```

Comment:
Should this be declared in the nested "if -> else -> if"?
The other place I see it being used (line 1389) seems to be overriding that value.

Reply:
`final_path_value` should live beyond the lifetime of `final_path` as it's the backing store for that stringview. So it cannot be declared within the scope of the `if`.